src/coreclr/jit/llvm.cpp

```diff
@@ -513,15 +567,20 @@ int getTotalParameterOffset(CORINFO_SIG_INFO& sigInfo)
 unsigned int getTotalRealLocalOffset()
 {
     unsigned int offset = 0;
-    // TODO: might need this IL->LLVM function for locals and exception funclets
-    // for (int i = 0; i < _locals.Length; i++)
-    //{
-    //    TypeDesc localType = _locals[i].Type;
-    //    if (!CanStoreVariableOnStack(localType))
-    //    {
-    //        offset = padNextOffset(localType, offset);
-    //    }
-    //}
+
+    for (unsigned int i = 0; i < _compiler->lvaTableCnt; i++)
+    {
+        LclVarDsc tableVar = _compiler->lvaTable[i];
+        if (!tableVar.lvIsParam)
+        {
+            CorInfoType corInfoType = toCorInfoType(tableVar.TypeGet());
+            if (!canStoreTypeOnLlvmStack(corInfoType))
+            {
+                offset = padNextOffset(corInfoType, offset);
+            }
+        }
+    }
```

Comment:
The "modern" way to write this loop would something like this:
```c++
for (unsigned lclNum = 0; lclNum < _compiler->lvaCount; lclNum++)
{
    LclVarDsc* varDsc = _compiler->lvaGetDesc(i);
}
```
Note that we never work with `LclVarDsc`s themselves, only `LclVarDsc*`, since they're pretty large (100+ bytes).